kafka/src/test/java/org/axonframework/extensions/kafka/eventhandling/consumer/streamable/ConsumerPositionsUtilIntegrationTest.java

```diff
@@ -90,31 +100,39 @@ void tearDown() {
         producerFactory.shutDown();
     }
 
-    @Test
-    void positionsTest() {
-        String topic = "testPositionsUtil";
+    <T extends TopicSubscriber> void _positionsTest(T subscriber) {
 
         Consumer<?, ?> testConsumer = consumerFactory.createConsumer(null);
-        List<String> topics = Collections.singletonList(topic);
-        assertTrue(ConsumerPositionsUtil.getHeadPositions(testConsumer, topics).isEmpty());
-        assertTrue(ConsumerPositionsUtil.getPositionsBasedOnTime(testConsumer, topics, Instant.now()).isEmpty());
+        assertTrue(ConsumerPositionsUtil.getHeadPositions(testConsumer, subscriber).isEmpty());
+        assertTrue(ConsumerPositionsUtil.getPositionsBasedOnTime(testConsumer, subscriber, Instant.now()).isEmpty());
 
         int recordsPerPartitions = 5;
         Producer<String, String> producer = producerFactory.createProducer();
-        publishRecordsOnPartitions(producer, topic, recordsPerPartitions, 5);
+        publishRecordsOnPartitions(producer, TEST_TOPIC, recordsPerPartitions, 5);
 
         Instant now = Instant.now();
-        publishRecordsOnPartitions(producer, topic, recordsPerPartitions, 5);
+        publishRecordsOnPartitions(producer, TEST_TOPIC, recordsPerPartitions, 5);
 
-        Map<TopicPartition, Long> headPositions = ConsumerPositionsUtil.getHeadPositions(testConsumer, topics);
+        Map<TopicPartition, Long> headPositions = ConsumerPositionsUtil.getHeadPositions(testConsumer, subscriber);
         assertFalse(headPositions.isEmpty());
         assertEquals(5, headPositions.keySet().size());
         headPositions.values().forEach(p -> assertEquals(9, p));
 
         Map<TopicPartition, Long> positionsBasedOnTime =
-                ConsumerPositionsUtil.getPositionsBasedOnTime(testConsumer, topics, now);
+                ConsumerPositionsUtil.getPositionsBasedOnTime(testConsumer, subscriber, now);
         assertFalse(positionsBasedOnTime.isEmpty());
         assertEquals(5, positionsBasedOnTime.keySet().size());
         positionsBasedOnTime.values().forEach(p -> assertEquals(4, p));
+
+    }
+
+    @Test
+    void positionsTestForListSubscriber() {
```

Comment:
Curious: why not use the `@ParameterizedTest` solution?

Reply:
sweet, i'll add that

Reply:
Pushed, however one sticking point is that I dropped the mock check for the subscribe() call. 
Example: `verify(mockConsumer, times(expectedNumberOfConsumers)).subscribe(Collections.singletonList(TEST_TOPIC));`
won't work for Pattern. 
I am thinking we could pass the test a class returns the builder, as well as does the verification. What do you think?